File: app/src/main/java/com/example/brauliomendez/cleanarquitecturedaggerandgithubapi/retrofit/GithubService.java
package com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.retrofit;

import com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.ApiConstants;
import com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.entity.SearchResponse;
import com.example.brauliomendez.cleanarquitecturedaggerandgithubapi.entity.User;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.POST;

/**
 * Created by BraulioMendez on 1/24/17.
 */

public interface GithubService {

    @POST(ApiConstants.USER_ENDPOINT)
    Call<User> getUserInfo(@Field("user_id") String id);

    @POST(ApiConstants.SEARCH_REPOSITORIES_ENDPOINT)
    Call<SearchResponse> getRepositories(@Field("keyword") String keyword);
}

File: kotlinversion/src/main/java/com/example/kotlinversion/client/GithubService.kt
package com.example.kotlinversion.client

import com.example.kotlinversion.entity.SearchResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by BraulioMendez on 1/29/17.
 */
interface GithubService {

    @GET(GithubClientConstants.SEARCH_REPOSITORIES_ENDPOINT)
    fun getRepositories(@Query("q") repositories : String) : Observable<SearchResponse>
}